internal/command/views/test.go

```diff
@@ -570,3 +569,61 @@ func testStatus(status moduletest.Status) string {
 		panic("unrecognized status: " + status.String())
 	}
 }
+
+// SaveErroredTestStateFile is a helper function to invoked in DestorySummary
+// to store the state to errored_test.tfstate and handle associated diagnostics and errors with this operation
+func SaveErroredTestStateFile(state *states.State, run *moduletest.Run, file *moduletest.File, view Test) {
+	var diags tfdiags.Diagnostics
+	localFileSystem := statemgr.NewFilesystem("errored_test.tfstate")
+	stateFile := statemgr.NewStateFile()
+	stateFile.State = state
+
+	//creating an operation to invoke EmergencyDumpState()
+	var op Operation
+	switch v := view.(type) {
+	case *TestHuman:
+		op = NewOperation(arguments.ViewHuman, false, v.view)
+		v.view.streams.Eprint(format.WordWrap("\nWriting state to file: errored_test.tfstate\n", v.view.errorColumns()))
+	case *TestJSON:
+		op = &OperationJSON{
+			view: v.view,
+		}
+		v.view.log.Info("Writing state to file: errored_test.tfstate")
```

Comment:
Here's why I think that `SaveErroredTestStateFile` belongs inside `internal/command/views/test.go` (as a part of the views package):
- The creation of `OperationJson`/`OperationHuman` requires the `view` field of `TestJson`/`TestHuman`,  whose scope is limited to this file.

- The function mostly performs view operations, by writing to the view using a stream. `EmergencyDumpState` is also a view operation, it dumps the statefile to the terminal. The actual file write is done by a call to `WriteStateForMigration`, which  is a part of the filesystem logic.